dapp/src/utils/contracts.js

```diff
@@ -128,16 +129,37 @@ export async function setupContracts(account, library, chainId) {
     }
   }
 
-  const callWithDelay = (fetchAPR = false) => {
+  const fetchCreditsPerToken = async () => {
+    const response = await fetch(process.env.CREDITS_ANALYTICS_ENDPOINT)
+    if (response.ok) {
+      const json = await response.json()
+      YieldStore.update((s) => {
+        s.currentCreditsPerToken = parseFloat(json.current_credits_per_token)
+        s.nextCreditsPerToken = parseFloat(json.next_credits_per_token)
+      })
+    }
+  }
+
+  const fetchCreditsBalance = async () => {
```

Comment:
this one in try-catch probably too.

So just in case any of these functions fail, that the other ones that get called after them still succeed.